execute/plugin.go

```diff
@@ -144,6 +156,17 @@ func getPendingExecutedReports(
 			continue
 		}
 
+		// Filter out reports that cannot be executed (snoozed).
+		// TODO: filter func instead of 'canExecute'?
+		var filtered []exectypes.CommitData
+		for _, commitReport := range reports {
```

Comment:
The original `commit_roots.go` also has a nice trick with a sliding window. Therefore we were limiting a number of database logs fetched from the db to memory by narrowing the lookup window based on the latest finalized and executed Commit Report. Not sure if applicable here, but I wanted to raise that as a follow-up perf improvement.

Reply:
Yes, thats another great optimization that I've been thinking about. We can take it a step further by adding a "Pending" cache, that way stuck transactions wouldn't prevent us from narrowing the lookup window.